src/tribler-core/tribler_core/session.py

```diff
@@ -220,48 +220,51 @@ def unhandled_error_observer(self, loop, context):
-            return
-
-        text_long = text
-        exc = context.get('exception')
-        if exc:
-            with StringIO() as buffer:
-                print_exception(type(exc), exc, exc.__traceback__, file=buffer)
-                text_long = text_long + "\n--LONG TEXT--\n" + buffer.getvalue()
-        text_long = text_long + "\n--CONTEXT--\n" + str(context)
-
-        description = 'session.unhandled_error_observer()'
-        with AllowSentryReports(value=False, description=description):
-            self._logger.error("Unhandled exception occurred! %s", text_long)
-
-        sentry_event = SentryReporter.last_event
-
-        if not self.api_manager:
-            return
-
-        events = self.api_manager.get_endpoint('events')
-        events.on_tribler_exception(text_long, sentry_event)
-
-        state = self.api_manager.get_endpoint('state')
-        state.on_tribler_exception(text_long, sentry_event)
+            exception = context.get('exception')
+            ignored_message = None
+            try:
+                ignored_message = IGNORED_ERRORS.get(
+                    (exception.__class__, exception.errno),
+                    IGNORED_ERRORS.get(exception.__class__))
+            except (ValueError, AttributeError):
+                pass
+            if ignored_message is not None:
+                self._logger.error(ignored_message if ignored_message != "" else context.get('message'))
+                return
+
+            text = str(exception or context.get('message'))
+
+            # We already have a check for invalid infohash when adding a torrent, but if somehow we get this
+            # error then we simply log and ignore it.
+            if isinstance(exception, RuntimeError) and 'invalid info-hash' in text:
+                self._logger.error("Invalid info-hash found")
+                return
+
+            text_long = text
+            exc = context.get('exception')
+            if exc:
+                with StringIO() as buffer:
+                    print_exception(type(exc), exc, exc.__traceback__, file=buffer)
+                    text_long = text_long + "\n--LONG TEXT--\n" + buffer.getvalue()
+            text_long = text_long + "\n--CONTEXT--\n" + str(context)
+
+            description = 'session.unhandled_error_observer()'
+            with AllowSentryReports(value=False, description=description):
+                self._logger.error("Unhandled exception occurred! %s", text_long)
+            sentry_event = SentryReporter.last_event
+
+            if not self.api_manager:
+                return
+
+            events = self.api_manager.get_endpoint('events')
+            events.on_tribler_exception(text_long, sentry_event)
+
+            state = self.api_manager.get_endpoint('state')
+            state.on_tribler_exception(text_long, sentry_event)
+        except Exception as e:
+            SentryReporter.send_exception_with_confirmation(e)
```

Comment:
This code looks like if some exception `e` were watched, it would be sent to Sentry. But, actually, `SentryReporter. send_exception_with_confirmation` uses `e` exception only for displaying it to the user. The exception that is actually sent is `SentryReporter.last_event`, which is not set here or inside `SentryReporter. send_exception_with_confirmation`.

Is it really true that `SentryReporter.last_event` is properly initialized here, does not equal to None, and actually corresponds to the `e` exception? The logic does not look straightforward to me.

As I understand it, `SentryReporter.last_event` becomes initialized as a side effect of the `before_send` hook that should be called when Sentry is trying to send some event. But does Sentry really sees the `e` exception? It was generated in our code, we catch it successfully, so if I understand correctly, Sentry will not have a chance to look at the exception and will not call the `before_send` hook as well.

Reply:
>  But does Sentry really sees the e exception?

Yes, it does.

Reply:
To make the function purpose more clear, its name has been changed.

Reply:
I will take the time to test it once more.

Reply:
Yes, you are right. I've changed  `SentryReporter.send_exception_with_confirmation(e)` to send this particular exception with no magic.